signer.go

```diff
@@ -109,12 +125,21 @@ func (s SharedSecretSigner) signStep(step reflect.Value) (interface{}, error) {
 		}
 	}
 
-	env[stepSignatureEnv] = commandSignature
+	env[stepSignatureEnv] = signature
 	copy["env"] = env
 
 	return copy, nil
 }
 
+func (s SharedSecretSigner) extractPlugins(plugins interface{}) (string, error) {
+	// TODO pre-process plugin sources per https://buildkite.com/docs/pipelines/plugins#plugin-sources
+	pluginJSON, err := json.Marshal(plugins)
```

Comment:
@lox to avoid issues with this JSON being re-ordered, I'm thinking to sort this (JSON) string before signing it. That way the order of the contents doesn't matter. Keen for thoughts

Reply:
Yup, I'd sort it.

Reply:
Awesome, will give that a go -- e.g. `canonicalizePlugins` or something